src/Middleware/HealthChecks/src/HealthCheckMiddleware.cs

```diff
@@ -55,6 +57,13 @@ public async Task InvokeAsync(HttpContext httpContext)
                 throw new ArgumentNullException(nameof(httpContext));
             }
 
+            if (!AllowedMethods.Contains(httpContext.Request.Method))
+            {
+                httpContext.Response.StatusCode = StatusCodes.Status405MethodNotAllowed;
+                httpContext.Response.Headers.Add(HeaderNames.Allow, AllowedMethods);
+                return;
```

Comment:
I'm not entirely certain about adding a 405 here. I know it **sounds** reasonable, but I'm not sure how we know for sure that no monitoring system does this.